### PR TITLE
[#1858] Use ttd as command consumer lifespan

### PR DESCRIPTION
**Pull request by calohmn**
This is for #1858.

### CODE REVIEW

File: client/src/main/java/org/eclipse/hono/client/ProtocolAdapterCommandConsumerFactory.java [hunk 1]
```diff
@@ -83,6 +85,7 @@ void initialize(CommandTargetMapper commandTargetMapper,
             String tenantId,
             String deviceId,
             Handler<CommandContext> commandHandler,
+            int lifespanSeconds,
```
Comment (ctron) [inline client/src/main/java/org/eclipse/hono/client/ProtocolAdapterCommandConsumerFactory.java:88]: I would prefer the use of `Duration` rather then `int`, which prevents any confusion around the unit.

File: client/src/main/java/org/eclipse/hono/client/ProtocolAdapterCommandConsumerFactory.java [hunk 2]
```diff
@@ -121,6 +126,7 @@ void initialize(CommandTargetMapper commandTargetMapper,
             String deviceId,
             String gatewayId,
             Handler<CommandContext> commandHandler,
+            int lifespanSeconds,
```
Comment (ctron) [inline client/src/main/java/org/eclipse/hono/client/ProtocolAdapterCommandConsumerFactory.java:129]: Same here.

File: adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/impl/VertxBasedAmqpProtocolAdapter.java [hunk 3]
```diff
@@ -864,10 +864,10 @@ private Span newSpan(final String operationName, final Device authenticatedDevic
         if (authenticatedDevice != null && !authenticatedDevice.getDeviceId().equals(sourceAddress.getResourceId())) {
             // gateway scenario
             return getCommandConsumerFactory().createCommandConsumer(sourceAddress.getTenantId(),
-                    sourceAddress.getResourceId(), authenticatedDevice.getDeviceId(), commandHandler, span.context());
+                    sourceAddress.getResourceId(), authenticatedDevice.getDeviceId(), commandHandler, -1, span.context());
```
Comment (ctron) [inline adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/impl/VertxBasedAmqpProtocolAdapter.java:867]: If the parameter is optional, then maybe consider using `Optional`. Or maybe `null` when using `Duration` (see other comments on "int").

File: client/src/main/java/org/eclipse/hono/client/ProtocolAdapterCommandConsumerFactory.java (hunk 1 above)
Comment (calohmn) [reply client/src/main/java/org/eclipse/hono/client/ProtocolAdapterCommandConsumerFactory.java]: I've amended the commit, changing the type to `Duration` (also in the `DeviceConnectionInfo`, `DeviceConnectionClient` and `DeviceConnectionService` interfaces committed earlier).
I've opted against the usage of `Optional` (it produces quite some clutter here, IMHO). The way to indicate an infinite lifespan is now using either `null` or a negative `Duration`.

File: adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/impl/VertxBasedAmqpProtocolAdapter.java (hunk 3 above)
Comment (calohmn) [reply adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/impl/VertxBasedAmqpProtocolAdapter.java]: See the other comment.

**Pull request merged**